plugins/flytekit-greatexpectations/flytekitplugins/great_expectations/task.py

```diff
@@ -0,0 +1,287 @@
+
+        # FlyteSchema
+        if type(dataset) is FlyteSchema:
+            # copy parquet file to user-given directory
+            FlyteContext.current_context().file_access.get_data(
+                dataset.remote_path, self._local_file_path, is_multipart=True
+            )
+
+        # DataFrame (Pandas, Spark, etc.)
+        else:
+            if not os.path.exists(self._local_file_path):
+                os.makedirs(self._local_file_path, exist_ok=True)
+
+            schema = FlyteSchema(
+                local_path=self._local_file_path,
+            )
+            writer = schema.open(type(dataset))
+            writer.write(dataset)
+
+        dataset = os.path.basename(self._local_file_path)
+
+        return dataset
+
+    def execute(self, **kwargs) -> Any:
+        context = ge.data_context.DataContext(self._context_root_dir)
+
+        if len(self.python_interface.inputs.keys()) != 1:
+            raise TypeError("Expected one input argument to validate the dataset")
+
+        dataset_key = list(self.python_interface.inputs.keys())[0]
+        dataset = kwargs[dataset_key]
+        datatype = self.python_interface.inputs[dataset_key]
+
+        if not issubclass(datatype, (FlyteFile, FlyteSchema, str)):
+            raise TypeError("'dataset' has to have FlyteFile/FlyteSchema/str datatype")
+
+        # determine the type of data connector
+        selected_datasource = list(filter(lambda x: x["name"] == self._datasource_name, context.list_datasources()))
+
+        if not selected_datasource:
+            raise ValueError("Datasource doesn't exist!")
+
+        data_connector_class_lookup = {
+            data_connector_name: data_connector_class["class_name"]
+            for data_connector_name, data_connector_class in selected_datasource[0]["data_connectors"].items()
+        }
+
+        specified_data_connector_class = data_connector_class_lookup[self._data_connector_name]
+
+        is_runtime = False
+        if specified_data_connector_class == "RuntimeDataConnector":
+            is_runtime = True
+            if not self._data_asset_name:
+                raise ValueError("data_asset_name has to be given in a RuntimeBatchRequest")
+
+        # FlyteFile
+        if issubclass(datatype, FlyteFile):
+            dataset = self._flyte_file(dataset)
+
+        # FlyteSchema
+        # convert schema to parquet file
+        if issubclass(datatype, FlyteSchema) and not is_runtime:
+            dataset = self._flyte_schema(dataset)
+
+        # minimalistic batch request
+        final_batch_request = {
+            "data_asset_name": self._data_asset_name if is_runtime else dataset,
+            "datasource_name": self._datasource_name,
+            "data_connector_name": self._data_connector_name,
+        }
+
+        # Great Expectations' RuntimeBatchRequest
+        if self._batch_request_config and (self._batch_request_config.runtime_parameters or is_runtime):
```

Comment:
is this always a json? not a typed structure?

Reply:
`_batch_request_config` is a dataclass, whereas `final_batch_request` is a dictionary.